program/Project Descriptions and Plans/CORE-V-HAL/Project-Concept-HAL.md

```diff
@@ -0,0 +1,155 @@
+
+
+| Category     	  		| Req #		| Requirement                  |
+|----------	  		|------		|-----------------------------	|
+| *Technical Reqmtns*	| T-1     	|HAL MUST support API for core features | 
+| 			  	| T-2     	|HAL MUST support API for on-chip peripheral features | 
+| 			  	| T-3     	|HAL MUST support both RTOS and bare-metal scenarios |
+| 			  	| T-4     	|HAL MUST be able to support bare metal libc |
+| 			  	| T-5     	|HAL MUST be able to be extended to support all CORE-V cores for the purpose of RTOS (not Linux) support |
+| 			  	| T-6     	|HAL MUST support a description of the chip using a standardized syntax |
+| 			  	| T-7     	|HAL SHOULD define a configuration file that can indicate to the IDE low level chip parameters such as memory space and allow it to display/read/write the registers of peripherals |
+| 			  	| T-8     	|HAL MAY be able to be extended to support the CORE-V X Coprocessor Interface (CV-X-IF) |
+| 			  	| T-9     	|HAL MAY be able to be extended to support the CV-VEC coprocessor | 
+| *Industry Adoption Rqmtns*	| IA-1     	|HAL SHOULD be adopted/implemented for other RISC-V devices | 	
+| *Specification Rqmtns*	| S-1     	|HAL Specification License for CORE-V MUST be open-source or freely licensed | 			  	| T-2     		| 
+| 				| S-2     	|Proprietary extensions to the HAL Specification and Implementation, such as for custom ISA extensions or proprietary chip, MUST be supportable/allowable under open-source license|
+| 				| S-3     	|HAL Specification, user documentation, and system implementation documentation for CORE-V HAL MUST be available in English |
+| 				| S-4     	|HAL Specification MUST be published and managed by an industry open source or standards body, such as OpenHW, RISC-V International, or FOSSi |
+| *Implementation Rqmtns* 	| IM-1     	|A complete open source reference implementation of the HAL for CORE-V MUST be a project deliverable |
+| 				| IM-2     	|FreeRTOS SHOULD be supported in the open source reference implementation of HAL for CORE-V |
+| 				| IM-3     	|Newlib SHOULD be supported in the open source reference implementation  of HAL for CORE-V |
+| 				| IM-3     	|The HAL reference implementation for CORE-V SHOULD be published by the body publishing the Specification (previous page) |
+  	
+
+
+### Potential future design enhancements
+- Support new peripherals or other Hardware components.
+- Create and provide reference designs on application areas such as Neural Network, DSP, Debug & Trace and Security.
+
+## Who would make use of OpenHW output
+CORE-V HAL will be used by firmware, middleware and application developers who are developing code supporting Core-V based hardware designs.
+
+It will be a key component of a Core-V MCU SDK.
+
+ 
+## Initial Estimate of Timeline
+The project will be last for 3 Months and have below output at different stage.
```

Comment:
This needs rewording. Arm CMSIS has been in development for 15 years.

I suggest something like:

"HAL's take many years to develop. Arm CMSIS is around 15 years old.  At this stage we propose a 3-month period to agree the approach and develop detailed plans for project launch.  At that time we shall have a more accurate understanding of the deliverables and the timescale over which they can be completed.

Reply:
Agree with Jeremy. I think the first step for us to make a detailed plan.